src/main/resources/additional_properties.json

```diff
@@ -18,5 +18,266 @@
+                "apply_to" : {
+                  "$ref" : "#/$defs/apply_to"
+                },
+                "name" : {
+                  "type" : "string",
+                  "default" : null
+                },
+                "version" : {
+                  "type" : "integer",
+                  "default" : null
+                }
+              }
+            }
+          }
+        },
+        {
+          "type" : "object",
+          "required" : [ "tombstone_handling" ],
+          "additionalProperties" : false,
+          "properties" : {
+            "tombstone_handling" : {
+              "additionalProperties" : false,
+              "required" : [ "behaviour" ],
+              "properties" : {
+                "behaviour" : {
+                  "type" : "string",
+                  "default" : "warn",
+                  "enum" : [ "warn", "ignore", "fail" ]
+                }
+              }
+            }
+          }
+        },
+        {
+          "type" : "object",
+          "required" : [ "value_to_key" ],
+          "additionalProperties" : false,
+          "properties" : {
+            "value_to_key" : {
+              "additionalProperties" : false,
+              "required" : [ "fields" ],
+              "properties" : {
+                "fields" : {
+                  "type" : "array",
+                  "items" : {
+                    "type" : "string"
+                  }
+                }
+              }
+            }
+          }
+        },
+        {
+          "type": "object",
+          "required": [ "topic_routing" ],
+          "additionalProperties": false,
+          "properties": {
+            "topic_routing": {
+              "additionalProperties": false,
+              "required": [ "topic_regex", "topic_replacement", "key_enforce_uniqueness", "key_field_name", "key_field_regex", "key_field_replacement" ],
+              "properties": {
+                "topic_regex": {
+                  "type": "string"
+                },
+                "topic_replacement": {
+                  "type": "string"
+                },
+                "key_enforce_uniqueness": {
+                  "type": "boolean",
+                  "default": true
+                },
+                "key_field_name": {
+                  "type": "string"
+                },
+                "key_field_regex": {
+                  "type": "string"
+                },
+                "key_field_replacement": {
+                  "type": "string"
+                }
+              }
+            }
+          }
+        },
+        {
+          "type" : "object",
+          "required" : [ "record_state_extraction" ],
+          "additionalProperties" : false,
+          "properties" : {
+            "record_state_extraction" : {
+              "additionalProperties" : false,
+              "required" : [ "drop_tombstones", "delete_handling_mode", "add_fields_prefix", "add_fields", "add_headers_prefix", "add_headers" ],
```

Comment:
`drop_tombstones` and `delete_handling_mode` has defaults and aren't mandatory properties.

Reply:
Does the non-mandatory option go to `additionalProperties` field?